### PR TITLE
ENH: improved clip function

### PR DESCRIPTION
**Pull request by eilidhmacnicol**
Using the improved clip function in niworkflows, the Laplacian filter normalisation and boundaries are more similar between images. 

Also used this point to update the b-spline grid and making it relative to image shape and zooms, and not just shape.

### CODE REVIEW

**Comment by eilidhmacnicol** [comment]
This PR requires testing on more data - @oesteban do you have time to run this branch on the original dataset or would it be easy enough to host it somewhere so I can download it and set it running via docker/singularity?

File: setup.cfg [hunk 1]
```diff
@@ -25,7 +25,7 @@ install_requires =
     nipype ~= 1.5.1
     niworkflows >= 1.3.2
     templateflow ~= 0.7.1
-    svgutils == 0.3.1
+    svgutils >= 0.3.4
```
Comment (oesteban) [inline setup.cfg:28]:
```suggestion
```

File: setup.cfg [hunk 2]
```diff
@@ -25,7 +25,7 @@ install_requires =
     nipype ~= 1.5.1
     niworkflows >= 1.3.2
```
Comment (oesteban) [inline setup.cfg:26]:
```suggestion
    niworkflows >= 1.4.0rc5
```

Although, this is going to bring a whole host of API changes, probably.

**Review by oesteban — COMMENTED** [review]
I think this goes in the right direction :)

File: setup.cfg (hunk 1 above)
Comment (oesteban) [reply setup.cfg:28]: Let niworkflows handle this, now that the bug is solved.

File: setup.cfg (hunk 2 above)
Comment (eilidhmacnicol) [reply setup.cfg]: Well, better to discover them now...

**Comment by eilidhmacnicol** [comment]
No issues with API after updating niworkflows dependency. CircleCI output isn't great, but running locally omitting the `--sloppy` command seems to output decent masking. Any outstanding issues or should I merge?

File: nirodents/workflows/brainextraction.py [hunk 3]
```diff
@@ -156,14 +146,11 @@ def init_rodent_brain_extraction_wf(
     wf = pe.Workflow(name)
 
     # truncate target intensity for N4 correction
-    clip_target = pe.Node(niu.Function(function=_trunc), name="clip_target")
-    clip_target.inputs.percentiles = (None, 99.9)
-    clip_target.inputs.clip_max = None
+    clip_target = pe.Node(IntensityClip(p_min=0, p_max=99.9), name="clip_target")
```
Comment (oesteban) [inline nirodents/workflows/brainextraction.py:149]:
```suggestion
    clip_target = pe.Node(IntensityClip(p_min=15.0, p_max=95.0), name="clip_target")
```

File: nirodents/workflows/brainextraction.py [hunk 4]
```diff
@@ -156,14 +146,11 @@ def init_rodent_brain_extraction_wf(
...
 
     # truncate template intensity to match target
-    clip_tmpl = pe.Node(niu.Function(function=_trunc), name="clip_tmpl")
+    clip_tmpl = pe.Node(IntensityClip(p_min=35, p_max=90), name="clip_tmpl")
```
Comment (oesteban) [inline nirodents/workflows/brainextraction.py:152]:
```suggestion
    clip_tmpl = pe.Node(IntensityClip(p_min=5, p_max=98), name="clip_tmpl")
```

**Review by oesteban — APPROVED** [review]
LGTM - I've made a couple of suggestions that I'm confident will help in this case

File: nirodents/workflows/brainextraction.py (hunk 3 above)
Comment (oesteban) [reply nirodents/workflows/brainextraction.py:149]: perhaps more conservatively:

```suggestion
    clip_target = pe.Node(IntensityClip(p_min=5, p_max=98), name="clip_target")
```

(like the template)

Comment (eilidhmacnicol) [reply nirodents/workflows/brainextraction.py:149]: After doing some tests locally, the above suggestions weren't quite right (at least not for the test data that runs on circleCI. These (compromise) settings worked well for the target: 
```suggestion
    clip_target = pe.Node(IntensityClip(p_min=15, p_max=99.9), name="clip_target")
```

File: nirodents/workflows/brainextraction.py (hunk 4 above)
Comment (eilidhmacnicol) [reply nirodents/workflows/brainextraction.py:152]: These values worked well with the template though

File: nirodents/workflows/brainextraction.py (hunk 3 above)
Comment (eilidhmacnicol) [reply nirodents/workflows/brainextraction.py:149]: I suppose we'll only find out if it's overfitting to one image once we test on more than one image...

**Comment by eilidhmacnicol** [comment]
Okay, as before, the CircleCI mask doesn't look the best, but running locally with these changes without ``--sloppy`` looks good. Merging for now.

**Pull request merged**